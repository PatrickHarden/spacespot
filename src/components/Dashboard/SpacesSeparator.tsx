import React from 'react'
import { useIntl } from 'react-intl'
import styled from 'styled-components'

const Container = styled.div`
  width: 100%;
  text-align: left;
  position: relative;
`

const SpacesCount = styled.span`
  line-height: 1.5em;
  color: black;
  font-size: 16px;
  font-weight: 400;
  background-color: white;
  padding: 0 25px;
  margin-left: 20px;
`
const Line = styled.div`
  :after {
    content: ' ';
    position: absolute;
    top: 60%;
    left: 50%;
    width: 100%;
    border: 1px solid #979797;
    border-top: 0;
    border-radius: 4px;
    box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.05);
    transform: translateX(-50%);
    z-index: -2;
  }
`

const SpacesSeparator = (props: { num: number }) => {
  const { formatMessage } = useIntl()
  const tn = (s: string, value: number) =>
    formatMessage({ id: s }, { number: value })
  const { num } = props
  return (
    <>
      <Container>
        <Line>
          <SpacesCount>{tn('DASHBOARD_NUM_SPACES', num)}</SpacesCount>
        </Line>
      </Container>
    </>
  )
}
export default SpacesSeparator
