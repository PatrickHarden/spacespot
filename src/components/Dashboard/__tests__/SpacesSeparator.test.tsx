import React from 'react'
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'

import SpacesSeparator from '../SpacesSeparator'

jest.mock('react-intl', () => ({
  ...jest.requireActual('react-intl'),
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockDispatch = jest.fn()
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}))

it('renders without crashing', () => {
  const wrapper = mount(<SpacesSeparator num={1} />)
  expect(mountToJson(wrapper)).toMatchSnapshot()
})
