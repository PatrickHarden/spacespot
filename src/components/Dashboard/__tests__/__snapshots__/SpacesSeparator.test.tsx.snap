// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<SpacesSeparator
  num={1}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "eUnCtI",
            "rules": Array [
              "
  width: 100%;
  text-align: left;
  position: relative;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa eUnCtI"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "xOUHL",
                  "rules": Array [
                    "
  :after {
    content: ' ';
    position: absolute;
    top: 60%;
    left: 50%;
    width: 100%;
    border: 1px solid #979797;
    border-top: 0;
    border-radius: 4px;
    box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.05);
    transform: translateX(-50%);
    z-index: -2;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat xOUHL"
            >
              <styled.span>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bwzfXH",
                        "isStatic": false,
                        "lastClassName": "dfhtkd",
                        "rules": Array [
                          "
  line-height: 1.5em;
  color: black;
  font-size: 16px;
  font-weight: 400;
  background-color: white;
  padding: 0 25px;
  margin-left: 20px;
",
                        ],
                      },
                      "displayName": "styled.span",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bwzfXH",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <span
                    className="sc-bwzfXH dfhtkd"
                  >
                    DASHBOARD_NUM_SPACES
                  </span>
                </StyledComponent>
              </styled.span>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</SpacesSeparator>
`;
